text/trace/0170-sampling-probability.md

```diff
@@ -0,0 +1,677 @@
+
+```
+P(span sampled) = I
+P(parent sampled) = H
+P(span sampled | parent sampled) = 1
+P(span sampled | parent not sampled) = D
+```
+
+Using the formula above, 
+
+```
+I = 1*H + D*(1-H)
+```
+
+solve for D:
+
+```
+D = (I - H) / (1 - H)
+```
+
+Now the Sampler makes a decision with probability `D`.  Whether the
+decision is true or false, propagate `I` as the new head inclusion
+probability.  If the decision is true, begin recording a sub-rooted
+trace with adjusted count `1/I`.
+
+## Proposed specification text
+
+The following text will be added to the semantic conventions for
+recording the Sampler name and adjusted count (if known) as
+OpenTelemetry Span attributes.
+
+```
+# Semantic conventions for Sampled spans
+
+This document defines how to describe an what sampling was performed
+when recording a span that has had sampling logic applied.
+
+Span sampling attributes support computing metrics about spans that
+are part of a sampled trace from knowing their sampling inclusion
+probability.
+
+The _adjusted count_ of a span is defined as follows:
+
+- Adjusted count equals zero when inclusion probability equals zero
+- Adjusted count equals the mathematical inverse (i.e., reciprocal) of sampling inclusion probability when inclusion probability is non-zero.
+
+Consumers of spans carrying an adjusted count attribute are able to
+use the adjusted count of the span to increment a counter of matching
+spans.
+
+## Probability Sampling Attributes
+
+The `sampler.adjusted_count` attribute MUST reflect an unbiased
+estimate of the number of representative spans in the population of
+spans being produced.
+
+When built-in Samplers are used, the name of the effective Sampler
+that computed the adjusted count is included to indicate how the sample
+was computed, which may give additional information.
+
+| Attribute | Type | Description | Examples | Required |
+|---------- | ---- | ----------- | -------- | -------- |
+| `sampler.adjusted_count` | number | Effective count of the span. | 10       | No  |
+| `sampler.name`           | string | The name of the Sampler.     | `Parent` | Yes |
+
+For the built-in samplers, the following names are specified:
+
+| Built-in Sampler | Sets `sampler.adjusted_count`? | `sampler.name` | Notes  |
+| ---------------- | ------------------------------ | -------------- | ------ |
+| AlwaysOn     | No    | Not applicable | Sampling attributes are not used    | 
+| AlwaysOff    | No    | Not applicable | Spans are not recorded              |
+| ParentBased  | Maybe | `Parent`       | Adjusted count requires propagation |
+| TraceIDRatio | Yes   | `TraceIDRatio` |                                     |
+```
+
+Note that the `AlwaysOn` and `AlwaysOff` Samplers do not need to
+record their names, since they are indistinguishable from not having
+a Sampler configured.  When there is no `sampler.name` attribute
+present and a Span is recorded, it should be counted as one span
+(i.e., count == adjusted_count).
```

Comment:
I find this unnecessarily complicated / confusing. Too many exceptions. I think a simpler approach is
  1. to have AlwaysOn sampler record count of 1, so that spans can be processed normally, without special-casing it,
  2. the absence of the adjusted_count attr to mean the weight is unknown, special processing is needed
  3. I would prefer `sampler.name=Parent` to be implied if not specified, since it's the most common behavior (but I realize it in itself is an exception)

Reply:
Here is the general rule I am aiming for:

**The absence of `sampler.adjusted_count` implies either adjusted count is 1 or is unknown, depending on whether `sampler.name` is defined.**

If `sampler.name` is defined and `sampler.adjusted_count` is not, you must have an external way to find the adjusted count (if you care--this will become legacy behavior). OTel defines only one such external mechanism (i.e., how to get the adjusted count when `sampler.name=Parent`).

The rationale behind omitting "adjusted count" when it equals 1 is that it is an adjustment to the natural count of one individual Span. There is no adjustment when the count is 1, so no reason to record an adjusted count. This implies that we can omit the `sampler.name` attribute when the count is known to equal 1 because it is equivalent to no sampling. 

> since it's the most common behavior

The default Sampler is `ParentBased(root=AlwaysOn)`. Given the specification I am proposing, the default SpanData will not include `sampler.name` or `sampler.adjusted_count` attributes, because all spans have adjusted count of 1 in a `ParentBased(root=AlwaysOn)` configuration.

Reply:
I have revised the text in order to present the rationale in a better way.